src/lib/languages/mod.rs

```diff
@@ -3,30 +3,22 @@ mod javascript;
 use super::*;
 pub use javascript::JavaScript;
 
-#[derive(Debug)]
-pub struct LangError {
-  language: String
-}
-
 pub enum Lang {
   JS
 }
 
-impl Into<&'static str> for Lang {
-  fn into(self) -> &'static str {
-    match self {
-      Self::JS => "JavaScript"
-    }
+impl Into<Lang> for String {
+  fn into(self) -> Lang {
+    return Lang::JS;
   }
 }
 
-
-pub fn generate(parser: Parser, lang: Lang) -> Result<String, String> {
-  let code = match lang {
+pub fn generate(parser: Parser, lang: String) -> Result<String, ParsingErrorType> {
+  let code = match lang.into() {
```

Comment:
I think this is a bit better, this way you can also provide just a `Lang::...` or x other type as long as they implement `Into<Lang>`
```suggestion
pub fn generate(parser: Parser, lang: Into<Lang>) -> Result<String, ParsingErrorType> {
  let code = match lang.into() {
```

Reply:
This seems to have broken the code and it won't allow the Into<Lang>.